### PR TITLE
chore: update main

### PR DESCRIPTION
**Pull request by axejintao**
Merge staging into main

Co-authored-by: default <bokwandu@gmail.com>
Co-authored-by: Bo Du <bo@chronosphere.io>
Co-authored-by: default <harlem.shakeshack@gmail.com>
Co-authored-by: Kaneki <79668923+ethkaneki@users.noreply.github.com>
Co-authored-by: Kaneki <ethkaneki@protonmail.com>

### CODE REVIEW

File: src/chains/strategies/bsc.strategy.ts [hunk 1]
```diff
@@ -1,7 +1,11 @@
 import { BadRequest, UnprocessableEntity } from '@tsed/exceptions';
 import { ethers } from 'ethers';
 import { TOKENS } from '../../config/constants';
+<<<<<<< HEAD
 import { getTokenPrice, getVaultTokenPrice, getWrapperTokenPrice } from '../../prices/prices-util';
+=======
+import { getTokenPrice } from '../../prices/prices-util';
+>>>>>>> 2e7d652 (feat: initial v2 working implementation (#52))
```
Comment (lookfirst) [inline src/chains/strategies/bsc.strategy.ts:8]: Fork is a great git client. =)

File: src/config/chain/chain.ts [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+
+export const supportedChains = [eth, bsc];
+
+/**
+ * Resolve a request query param chain lookup.
+ * Failure to specify a chain, or requesting a non-supported
+ * chain will default to Ethereum.
+ */
+export const resolveChainQuery = (symbol?: string): Chain => {
+  if (!symbol) return eth;
```
Comment (lookfirst) [inline src/config/chain/chain.ts:51]: That feels really strange.

File: src/interface/Sett.ts [hunk 3]
```diff
@@ -1,7 +1,13 @@
 import { Protocol } from '../config/constants';
+<<<<<<< HEAD
```
Comment (lookfirst) [inline src/interface/Sett.ts:2]: so much merge...

File: src/config/chain/chain.ts (hunk 2 above)
Comment (axejintao) [reply src/config/chain/chain.ts:51]: this is for backward compatibility / no chain query params

File: src/chains/strategies/bsc.strategy.ts (hunk 1 above)
Comment (axejintao) [reply src/chains/strategies/bsc.strategy.ts]: yeah might look into it. this particular one had like probaly 40 conflicts, so i missed a few as i went through by folder

**Pull request closed without merging**